test_vm/tests/commit_post_test.rs

```diff
@@ -153,3 +155,423 @@ fn commit_post_flow_happy_path() {
+
+    // Fail with proof too big
+
+    let just_right_sectors_no_bf =
+        precommited_sector_nos.slice(0, policy.max_aggregated_sectors).unwrap();
+    prove_params = ProveCommitAggregateParams {
+        sector_numbers: UnvalidatedBitField::from(just_right_sectors_no_bf),
+        aggregate_proof: vec![0; policy.max_aggregated_proof_size + 1],
+    };
+
+    prove_params_ser = serialize(&prove_params, "commit params").unwrap();
+    apply_code(
+        &v,
+        worker,
+        robust_addr,
+        TokenAmount::zero(),
+        MinerMethod::ProveCommitAggregate as u64,
+        prove_params,
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+    );
+    ExpectInvocation {
+        to: id_addr,
+        method: MinerMethod::ProveCommitAggregate as u64,
+        params: Some(prove_params_ser),
+        subinvocs: Some(vec![]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+}
+
+#[test]
+fn aggregate_bad_sender() {
+    let store = MemoryBlockstore::new();
+    let mut v = VM::new_with_singletons(&store);
+    let addrs = create_accounts(&v, 2, TokenAmount::from(10_000e18 as i128));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (owner, worker) = (addrs[0], addrs[0]);
+    let (id_addr, robust_addr) = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from(10_000e18 as i128),
+    );
+    let mut v = v.with_epoch(200);
+    let policy = &Policy::default();
+
+    //
+    // precommit good sectors
+    //
+
+    // precommit and advance to prove commit time
+    let sector_number: SectorNumber = 100;
+    let mut precommited_sector_nos = BitField::try_from_bits(
+        precommit_sectors(
+            &mut v,
+            4,
+            policy.pre_commit_sector_batch_max_size as i64,
+            worker,
+            id_addr,
+            seal_proof,
+            sector_number,
+            true,
+            None,
+        )
+        .iter()
+        .map(|info| info.info.sector_number),
+    )
+    .unwrap();
+
+    //
+    // attempt proving with invalid args
+    //
+
+    // advance time to max seal duration
+
+    let prove_time = v.get_epoch() + max_prove_commit_duration(policy, seal_proof).unwrap();
+    let v = advance_by_deadline_to_epoch(v, id_addr, prove_time).0;
+
+    // construct invalid bitfield with a non-committed sector number > abi.MaxSectorNumber
+
+    precommited_sector_nos.set(MAX_SECTOR_NUMBER + 1);
+
+    let prove_params = ProveCommitAggregateParams {
+        sector_numbers: UnvalidatedBitField::from(precommited_sector_nos),
+        aggregate_proof: vec![],
+    };
+    let prove_params_ser = serialize(&prove_params, "commit params").unwrap();
+    apply_code(
+        &v,
+        addrs[1],
+        robust_addr,
+        TokenAmount::zero(),
+        MinerMethod::ProveCommitAggregate as u64,
+        prove_params,
+        ExitCode::USR_FORBIDDEN,
```

Comment:
Note that this is a different code than go-actors, which returns SysErrForbidden (8).

This makes sense based on the change introduced [here](https://github.com/filecoin-project/builtin-actors/commit/293ceb4853513265e5c5d3b4ff7659d4b4102b05#diff-6caa79c5439bd662a6215efdc4a8dac6bbbccd42a9e8c50994665b261d0a52d9R124), but it woudl be nice to get @Stebalien to explicitly :+1:

Reply:
Yes, this is intentional. The actor can't return a system exit code.